src/Kmd.Logic.Gateway.Automation/PreValidation/ProductPreValidation.cs

```diff
@@ -0,0 +1,50 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Security.Cryptography.X509Certificates;
+using System.Text;
+using System.Threading.Tasks;
+using Kmd.Logic.Gateway.Automation.Gateway;
+
+namespace Kmd.Logic.Gateway.Automation.PreValidation
+{
+    public class ProductPreValidation : EntityPreValidationBase, IValidation
+    {
+        public ProductPreValidation(string folderPath)
+           : base(folderPath)
+        {
+        }
+
+        public Task<bool> ValidateAsync(GatewayDetails gatewayDetails)
+        {
+            var isValidationSuccess = true;
+            if (gatewayDetails != null)
+            {
+                foreach (var product in gatewayDetails.Products)
+                {
+                    if (string.IsNullOrEmpty(product.Name))
+                    {
+                        this.ValidationResults.Add(new PublishResult { IsError = true, ResultCode = ResultCode.InvalidInput, Message = $"Product Name not exist" });
+                        isValidationSuccess = false;
+                    }
+
+                    if (!this.ValidateFile(GatewayFileType.Logo, product.Logo, $"{product.Name} ", nameof(product.Logo)))
```

Comment:
Do you need string interpolation here? `$"{product.Name} "`

Reply:
Not required